sopht_simulator/immersed_body/cosserat_rod/CosseratRodIO.py

```diff
@@ -0,0 +1,31 @@
+from sopht.utils.IO import IO
+from elastica.rod.cosserat_rod import CosseratRod
+import numpy as np
+
+
+class CosseratRodIO(IO):
+    def __init__(self, cosserat_rod: type(CosseratRod), dim, real_dtype=np.float64):
+        super().__init__(dim, real_dtype)
+        self.cosserat_rod = cosserat_rod
+
+        # Initialize rod element position
+        self.rod_element_position = np.zeros((3, cosserat_rod.n_elems))
+        self._update_rod_element_position
+
+        # Add the element position to IO
+        self.add_as_lagrangian_fields_for_io(
+            lagrangian_grid=self.rod_element_position,
+            lagrangian_grid_name="rod",
+            scalar_3d=self.cosserat_rod.radius,
+            lagrangian_grid_connect=True,
+        )
+
+    def save_rod(self, h5_file_name, time=0.0):
+        self._update_rod_element_position()
+        self.save(h5_file_name=h5_file_name, time=time)
+
+    def _update_rod_element_position(self):
```

Comment:
This is IO stuff, please put it in `utils/io_utils.py` instead of immersed body.

Reply:
I was thinking of putting it in there as well, but since this is a Cosserat rod concept, I placed it here. I was thinking along the lines of since we have `rigid_body` and `cosserat_rod` in `immersed_body`, one can also have other types of immersed bodies in the future, and perhaps the IO utils for each of these immersed bodies should be in `immersed_body` module. I guess since for now we only have Cosserat rod, I'll go ahead and move it as you suggested, but maybe worth revisiting when the collection of immersed bodies expands.

Fixed in [722f451](https://github.com/SophT-Team/SophT-Simulator/pull/70/commits/722f451c2f4e0b8a43da93ede8af488d35a7eba5)